Use it for the `zigzag` example and document it in the `README#usage`

Assignee: @schomatis.

Related to https://github.com/filecoin-project/rust-proofs/issues/481.

The CPU profiler works, trying to enable the heap profiler now the compiler fails to find the `heap_profiler` import although the IDE seems able to find it:

```
error[E0432]: unresolved import `gperftools::heap_profiler`
  --> filecoin-proofs/examples/zigzag.rs:19:17
   |
19 | use gperftools::heap_profiler::HEAP_PROFILER;
   |                 ^^^^^^^^^^^^^ could not find `heap_profiler` in `gperftools`
```

Looking into it.

--- a/README.md
+++ b/README.md
@@ -75,6 +75,8 @@ To use the heap profiler, a custom allocator is used. This feature requires sett
 
 You can find additional documentation about heap profiling [here](https://gperftools.github.io/gperftools/heapprofile.html).
 
+For a reproducible example of the heap profiler used on the [`rust-fil-proofs`](https://github.com/filecoin-project/rust-fil-proofs) repository see the [`rust-fil-proofs-profile`](https://hub.docker.com/r/schomatis/rust-fil-proofs-profile) Docker image.
+
 ### Analyzing the profile
 
 To analyze the profile use Google's [pprof](https://github.com/google/pprof) tool.

